src/app/content/components/popUp/types.ts

```diff
@@ -0,0 +1,16 @@
+import { LinkedArchiveTree, LinkedArchiveTreeNode, LinkedArchiveTreeSection } from '../../types';
+
+export type LocationFilters = Map<
+  string,
+  { section: LinkedArchiveTreeNode, children?: LinkedArchiveTreeSection[] }
+>;
+
+export type LocationFiltersWithChildren = Map<
+  string,
+  { section: LinkedArchiveTree, children: LinkedArchiveTreeSection[] }
+>;
+
+export interface FiltersChange<T> {
```

Comment:
i didn't catch why you changed the way the filters work to include the add/remove lists, was there some kind of race condition?

Reply:
I don't remember exactly (last commit 1,5 month ago)
There was something about updating the ChapterFilter where I need to receive the `section` instead of the `id` and since previously everything was working on checkboxes we were dispatching all selected ids + the newly clicked one
Anyway, I remember that when I was trying to implement a solution I thought that it would be cleaner if we would have a standardized way of handling Color a Chapter filter updates

Reply:
well we used to dispatch the new selected set, now you've made it dispatch a diff, the only reason i can think of that being helpful is if they are received out of order or something